src/Client.js

```diff
@@ -563,6 +563,40 @@ class Client extends EventEmitter {
         }, chatId);
     }
 
+    /**
+     * Enables and returns the pin state of the Chat
+     * @returns {boolean}
+     */
+    async pinChat(chatId) {
+        return await this.pupPage.evaluate(async chatId => {
+            let chat = await window.Store.Chat.get(chatId);
+            if (chat.pin) {
+                return true;
+            }
+            let maxPinned = await window.Store.Chat.models[2].pin;
```

Comment:
Could there be a better way to get the max number of pinned chats allowed / current number of pinned chats for this check?

This seems like it could be error prone if maybe there are less than 3 chats

Reply:
Yeah... I searched quite a bit. I'll try again with a fresh mind. But couldn't find anything promising.